src/services/EdFi.DataManagementService.Api.Tests.E2E/Features/ValidationErrors.feature

```diff
@@ -58,3 +58,84 @@ Scenario: Post a valid Descriptor
   """
   Then the response code is 201
 
+Scenario: Post an invalid body for academicWeeks missing more than one required field
+  When sending a POST request to "data/ed-fi/academicWeeks" with body
+  """
+  {
+    "_weekIdentifier": "abcdef",
+    "_schoolReference": {
+        "schoolId": 255901001
+    },
+    "_beginDate": "2024-04-04",
+    "_endDate": "2024-04-04",
+    "_totalInstructionalDays": 300
+  }
+  """
+  Then the response code is 400
+    And the response body is
+"""
+{"detail":"Data validation failed. See 'validationErrors' for details.","type":"urn:ed-fi:api:bad-request:data","title":"Data Validation Failed","status":400,"correlationId":null,"validationErrors":{"$.schoolReference":["schoolReference is required."],"$.weekIdentifier":["weekIdentifier is required."],"$.beginDate":["beginDate is required."],"$.endDate":["endDate is required."],"$.totalInstructionalDays":["totalInstructionalDays is required."]},"errors":[]}
+"""
+
+Scenario: Post an invalid body for academicWeeks missing a required field in a nested object schoolid for schoolReference
+  When sending a POST request to "data/ed-fi/academicWeeks" with body
+  """
+  {
+    "weekIdentifier": "abcdef",
+    "schoolReference": {
+        "_schoolId": 255901001
+    },
+    "beginDate": "2024-04-04",
+    "endDate": "2024-04-04",
+    "totalInstructionalDays": 300
+  }
+  """
+  Then the response code is 400
+    And the response body is
+"""
+{"detail":"Data validation failed. See 'validationErrors' for details.","type":"urn:ed-fi:api:bad-request:data","title":"Data Validation Failed","status":400,"correlationId":null,"validationErrors":{"$.schoolReference.schoolId":["schoolId is required."]},"errors":[]}
+"""
+
+Scenario: Post an invalid body for academicWeeks missing a comma before beginDate
+  When sending a POST request to "data/ed-fi/academicWeeks" with body
+  """
+  {
+    "weekIdentifier": "abcdef",
+    "schoolReference": {
+        "schoolId": 255901001
+    }
+    "beginDate": "2024-04-04",
+    "endDate": "2024-04-04",
+    "totalInstructionalDays": 300
+  }
+  """
+  Then the response code is 400
+    And the response body is
+"""
+{"detail":"Data validation failed. See 'validationErrors' for details.","type":"urn:ed-fi:api:bad-request:data","title":"Data Validation Failed","status":400,"correlationId":null,"validationErrors":{"$.":["'\"' is invalid after a value. Expected either ',', '}', or ']'. LineNumber: 5 | BytePositionInLine: 2."]},"errors":[]}
```

Comment:
This error detail differs a bit from the ODS example, here I am simply sending the exception I get in logginMiddleware

Reply:
There is a real problem with this error message though: it says that `\` is invalid after a value, but `\` is not in the payload. This makes me think that our code is not transmitting the payload correctly as shown.

Here's what the ODS/API says:

```json
{
  "detail": "Data validation failed. See 'validationErrors' for details.",
  "type": "urn:ed-fi:api:bad-request:data",
  "title": "Data Validation Failed",
  "status": 400,
  "correlationId": "ec5f955d-f8f8-431b-aaa3-a0e2a684136c",
  "validationErrors": {
    "$.schoolReference": [
      "After parsing a value an unexpected character was encountered: \". Path 'schoolReference', line 6, position 4."
    ]
  }
}
```

Its okay if our validation message is a little different in this case. It would be hard to have exactly the same, since we're using a different JSON parser.